src/snowflake/connector/pandas_tools.py

```diff
@@ -202,90 +192,85 @@ def write_pandas(
+        cursor.execute(file_format_sql, _is_internal=True)
 
-    if overwrite:
-        if auto_create_table:
-            drop_table_sql = f"DROP TABLE IF EXISTS {location} /* Python:snowflake.connector.pandas_tools.write_pandas() */ "
-            logger.debug(f"dropping table with '{drop_table_sql}'")
-            cursor.execute(drop_table_sql, _is_internal=True)
-        else:
-            truncate_table_sql = f"TRUNCATE TABLE IF EXISTS {location} /* Python:snowflake.connector.pandas_tools.write_pandas() */ "
-            logger.debug(f"truncating table with '{truncate_table_sql}'")
-            cursor.execute(truncate_table_sql, _is_internal=True)
-
-    if auto_create_table:
-        file_format_name = None
-        while True:
-            try:
-                file_format_name = (
-                    '"'
-                    + "".join(random.choice(string.ascii_lowercase) for _ in range(5))
-                    + '"'
-                )
-                file_format_sql = (
-                    f"CREATE FILE FORMAT {file_format_name} "
-                    f"/* Python:snowflake.connector.pandas_tools.write_pandas() */ "
-                    f"TYPE=PARQUET COMPRESSION={compression_map[compression]}"
-                )
-                logger.debug(f"creating file format with '{file_format_sql}'")
-                cursor.execute(file_format_sql, _is_internal=True)
-                break
-            except ProgrammingError as pe:
-                if pe.msg.endswith("already exists."):
-                    continue
-                raise
         infer_schema_sql = f"SELECT COLUMN_NAME, TYPE FROM table(infer_schema(location=>'@\"{stage_name}\"', file_format=>'{file_format_name}'))"
         logger.debug(f"inferring schema with '{infer_schema_sql}'")
-        column_type_mapping = dict(
-            cursor.execute(infer_schema_sql, _is_internal=True).fetchall()
-        )
-        # Infer schema can return the columns out of order depending on the chunking we do when uploading
-        # so we have to iterate through the dataframe columns to make sure we create the table with its
-        # columns in order
-        quote = '"' if quote_identifiers else ""
-        create_table_columns = ", ".join(
-            [f"{quote}{c}{quote} {column_type_mapping[c]}" for c in df.columns]
+        try:
+            column_type_mapping = dict(
+                cursor.execute(infer_schema_sql, _is_internal=True).fetchall()
+            )
+            # Infer schema can return the columns out of order depending on the chunking we do when uploading
+            # so we have to iterate through the dataframe columns to make sure we create the table with its
+            # columns in order
+            create_table_columns = ", ".join(
+                [f"{quote}{c}{quote} {column_type_mapping[c]}" for c in df.columns]
+            )
+        finally:
+            drop_object(file_format_name, "file format")
```

Comment:
It's a temp file format, I'd argue that we can save the round-trip time and have it be garbage collected by the server-side

Reply:
done